### PR TITLE
KFSPTS-28680 Update ISO 20022 msg ID and creditor handling

### PR DESCRIPTION
**Pull request by cah292**
This PR adds a few changes requested by the new bank:

* The file/message ID in the ISO 20022 file has to be unique over a period of a few weeks, as required by the bank. Base financials is just using the PDP Process ID as the file/message ID, resulting in both the Check file and the ACH file (and our custom Immediate Check file) having the same ID for a given PDP Process. To satisfy the uniqueness requirements, this PR appends an extra suffix to the file/message ID based on the types of transactions being processed.
* The bank does not want us to add the Creditor Reference Information section if the payment does not have a related Purchase Order. This PR updates the relevant code so that the section will only be included when the payment specifies a PO number.

### CODE REVIEW

**Review by nkimble — COMMENTED** [review]
Changes look appropriate. I had some suggestions if they make sense with respect to the present project requirements.

File: src/main/java/org/kuali/kfs/pdp/batch/service/impl/Iso20022FormatExtractor.java [hunk 1]
```diff
@@ -573,6 +578,23 @@ private GroupHeader32 constructGroupHeader(
         return groupHeader;
     }
 
+    /*
+     * CU Customization: Added method for deriving the suffix to append to the file's Message ID.
+     */
+    private String determineMessageIdSuffix(
+            final ExtractTypeContext extractTypeContext
+    ) {
```
[the remark this answers is not in the document]
Comment (cah292) [reply src/main/java/org/kuali/kfs/pdp/batch/service/impl/Iso20022FormatExtractor.java:586]: I was trying to have this method follow the same or similar formatting conventions as the other methods in this overlayed file. If needed, I could update the method to follow our cu-kfs conventions instead.

We could also discuss this at one of tomorrow's developer meetings, to discern when we should (or shouldn't) adjust new code to fit with what exists in the overlay, and to discern whether we need adjustments to our coding conventions in response to KualiCo's changes.

File: src/main/java/edu/cornell/kfs/pdp/CUPdpConstants.java [hunk 2]
```diff
@@ -63,5 +63,11 @@ public static class PayeeAchAccountExtractStatuses {
     public static class Iso20022Constants {
         public static final String LOCAL_INSTRUMENT_ISSUE_ONLY = "CII";
         public static final String ADJUSTMENT_REASON_NOTE_ONLY = "NOTE";
+
+        public static class MessageIdSuffixes {
+            public static final String ACH = "_A";
+            public static final String CHECK = "_C";
+            public static final String IMMEDIATE_CHECK = "_I";
+        }
```
[the remark this answers is not in the document]
Comment (cah292) [reply src/main/java/edu/cornell/kfs/pdp/CUPdpConstants.java:71]: Upon further consideration, I agree that the 3-letter suffixes without the underscore would work out better overall. That would still leave us with 3 digits in the last-6-chars range, which should be more than enough to keep things unique over the bank's required time period.

I'll adjust the constants accordingly.

**Comment by cah292** [comment]
I've committed some minor changes based on the initial feedback. Also, as per today's team meeting, I believe the formatting of that method declaration is okay for now.

This is ready for review again.

**Pull request merged**